client/app/scripts/superdesk-highlights/highlights.js

```diff
@@ -250,27 +250,67 @@
         };
     }
 
-    PackageHighlightsDropdownDirective.$inject = ['superdesk', 'desks', 'highlightsService'];
-    function PackageHighlightsDropdownDirective(superdesk, desks, highlightsService) {
+    PackageHighlightsDropdownDirective.$inject = ['superdesk', 'desks', 'highlightsService', '$location', '$route'];
+    function PackageHighlightsDropdownDirective(superdesk, desks, highlightsService, $location, $route) {
         return {
             templateUrl: 'scripts/superdesk-highlights/views/package_highlights_dropdown_directive.html',
             link: function(scope) {
-
-                scope.createHighlight = function(highlight) {
-                    highlightsService.createEmptyHighlight(highlight)
-                    .then(function(new_package) {
-                        superdesk.intent('author', 'package', new_package);
+                scope.$watch(function() {
+                    return desks.active;
```

Comment:
Yes, I'm sure, although it's probably fine in this case. :)

The thing is that this directive does have its own scope, it uses its parent element's scope. In theory it's possible that the directive element gets removed (for whatever reason), while the parent element is left intact and thus its scope would still be alive, along with the directive's watcher.

But as I said, it's probably just a theoretical scenario in this particular case, right?